### DISCUSSION TITLE
How to boot a Stratis rootfs?

### DISCUSSION THREAD

**Discussion by i2** [stratis-storage/project#539] [Q&A]
Hi everyone, I have a question in response to [this issue](https://github.com/stratis-storage/stratisd/issues/3019):

@refi64, @mulkieran, @jbaublitz and others: Would you please be able to tell me the exact steps you have followed to successfully install this on Stratis as rootfs?

I am trying to setup RHEL/Fedora on Stratis rootfs and have followed [these](https://stratis-storage.github.io/stratis-rootfs/) [two](https://stratis-storage.github.io/stratisd-rootfs-packaging/) guides. When I am done with setup and reboot, it drops me on GRUB's `minimal BASH-like` rescue terminal.

This is a EFI computer and my disk layout is:
```
/dev/sda1 -> EFI FAT32
/dev/sda2 -> rootfs for Stratis
```
Here's a template for my /mnt/etc/default/grub
```
GRUB_CMDLINE_LINUX="root=/dev/stratis/mypool/fs01 stratis.rootfs.pool_uuid=[POOL_UUID]"
```

And then I rebuild GRUB config file with above changes.

Is it not booting because GRUB can't find /boot/initramfs-XYZ.img which is on stratis rootfs? Do I need to have a separate non-Stratis boot partition which houses initramfs?

Thanks!

**Comment by jbaublitz** [comment]
A separate boot partition is indeed required. The pool set up code is in the initramfs so putting the initramfs on the pool would be a circular dependency.

**Reply by i2** [reply]
So the most minimal partitioning layout (for EFI) would be:


/dev/sda1 -> /boot/efi (EFI **FAT32**)
/dev/sda2 -> /boot (**ext4**)
/dev/sda3 -> rootfs for Stratis


Correct?

**Reply by jbaublitz** [reply]
That's correct.

**Reply by i2** [reply]
Thanks for confirming. I will continue my setup and report back if there are any further issues.

**Reply by i2** [reply]
I followed the setup through, created EFI entry for my install and when I reboot, it just hangs at:
```
md: Autodetecting RAID arrays.
md: autorun ...
md: ... autorun DONE.
```

**Reply by i2** [reply]
Like I mentioned above, I put `GRUB_CMDLINE_LINUX="root=/dev/stratis/mypool/fs01 stratis.rootfs.pool_uuid=[POOL_UUID]"` in `/mnt/etc/default/grub`

After generating grub.cfg, I open it up and go to the relevant menuentry. the kernel line is as follows:
```
linux /vmlinuz-XYZ root=/dev/mapper/stratis-1-XYZ-thin-fs-UUID ro root=/dev/stratis/mypool/fs01 stratis.rootfs.pool_uuid=UUID
```

As there were 2 **root=...** sections, I removed the first one (that is auto-generated) and only left the one I manually added, Then I rebooted. It still hangs at md: ... autorun DONE.

**Reply by jbaublitz** [reply]
It sounds like you have your Stratis pools on md RAID arrays, is that correct? We've had other reports of hanging boots when using RAID underneath Stratis but haven't been able to reproduce it yet. Does the boot time out eventually? If so, can you get the logs from the failed boot and provide them here?

**Reply by i2** [reply]
No, I do not have the pools on md RAID arrays. It actually surprised me to see those error!
Please let me know and I can provide more logs etc. to debug this issue. Thank you!

**Reply by jbaublitz** [reply]
Yes, please provide the logs as a starting point so I can see what's happening.
